### PR TITLE
feat: add updateNode/updateEdge API #3235

### PR DESCRIPTION
**Pull request by lloydzhou**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Enhancement (changes that improvement of current feature or performance)
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Test Case (changes that add missing tests or correct existing tests)
- [ ] Code style optimization (changes that do not affect the meaning of the code)
- [ ] Docs (changes that only update documentation)
- [ ] Chore (changes that don't modify src or test files)

### Self Check before Merge

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/antvis/x6/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by lloydzhou** [comment]
close #3235

**Comment by lloydzhou** [comment]
## 测试了一下：
1. updateNode(metadata)，如果metadata包含ports的时候，假如metadata.ports.items.length增加的时候画布能更新，metadata.ports.items.length减少的时候画布不能更新，必须手动调用一下cell.setProp('ports', metadata.ports)。
2. updateEdge(metadata)，如果包含labels的时候，和前面的表现一样不符合预期

还需要更完善的测试...

**Comment by lloydzhou** [comment]
> ## 测试了一下：
> 1. updateNode(metadata)，如果metadata包含ports的时候，假如metadata.ports.items.length增加的时候画布能更新，metadata.ports.items.length减少的时候画布不能更新，必须手动调用一下cell.setProp('ports', metadata.ports)。
> 2. updateEdge(metadata)，如果包含labels的时候，和前面的表现一样不符合预期
> 
> 还需要更完善的测试...

问题似乎出在直接setProp(prop)调用的时候，走的流程和setProp(key, value)流程不一致导致的

File: packages/x6/src/model/model.ts [hunk 1]
```diff
@@ -244,6 +244,13 @@ export class Model extends Basecoat<Model.EventArgs> {
     return node
   }
 
+  updateNode(metadata: Node.Metadata, options: Model.SetOptions = {}) {
+    const node = this.createEdge(metadata)
```
Comment (NewByVector) [inline packages/x6/src/model/model.ts:248]: createNode?

**Review by NewByVector — CHANGES_REQUESTED** [review]

File: packages/x6/src/model/model.ts (hunk 1 above)
Comment (lloydzhou) [reply packages/x6/src/model/model.ts]: 抱歉没有在x6项目这边直接写测试用例。
我只是在自己的另一个项目里面使用了类似的更新节点和边的代码进行测试。所以没注意到这个地方调用函数写错了。

**Review by NewByVector — APPROVED** [review]

**Pull request merged**